spec/controllers/profile_controller_spec.rb

```diff
@@ -13,23 +15,54 @@
       end
 
       it "get profile if have user" do
-        @request.env["devise.mapping"] = Devise.mappings[:user]
-        user = FactoryBot.create(:user_manager_user)
-        allow(request.env['warden']).to receive(:authenticate!).and_return(@user)
-        allow(controller).to receive(:current_user).and_return(@user)
-        sign_in user
+        usermgr_stub_auth
         get :show
         expect(response.status).to eq(200)
       end
     end
-    
+
+    context 'changing passwords' do
+      it 'errors when new passwords dont match' do
```

Comment:
this should be part of the model tests (which you don't have) and all combinations should be tested against the model. The controller test here can then test the cognito responses and behaviour...